### PR TITLE
feat: Add `ParseXFile` for cross-platform `XFile` support

### PR DESCRIPTION
**Pull request by mbfakourii**
## Pull Request

- Report security issues [confidentially](https://github.com/parse-community/Parse-SDK-Flutter/security/policy).
- Any contribution is under this [license](https://github.com/parse-community/Parse-SDK-Flutter/blob/master/LICENSE).
- Link this pull request to an [issue](https://github.com/parse-community/Parse-SDK-Flutter/issues?q=is%3Aissue).

## Issue
Add `ParseXFile` for support [`XFile`](https://pub.dev/documentation/cross_file/latest/cross_file/XFile-class.html).

Closes: #989

## Tasks

- [ ] Add tests
- [ ] Add changes to documentation (guides, repository pages, code comments)

### CODE REVIEW

**Comment by mtrezza** [comment]
Is this ready for merge?

**Comment by mbfakourii** [comment]
> Is this ready for merge?

Yes, I think it is ready to merge

**Comment by mbfakourii** [comment]
> Is this ready for merge?

Yes, I think it is ready to merge.

File: packages/dart/CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,9 @@
+## [6.4.0](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-6.3.0...dart-6.4.0) (2024-03-20)
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:1]:
```suggestion
## [6.4.0](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-6.3.0...dart-6.4.0) (2024-03-30)
```

**Review by mtrezza — DISMISSED** [review]
Looks good!

File: packages/dart/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,9 @@
+## [6.4.0](https://github.com/parse-community/Parse-SDK-Flutter/compare/dart-6.3.0...dart-6.4.0) (2024-03-30)
+
+### Features
+
+* Add `ParseXFile` ([#990](https://github.com/parse-community/Parse-SDK-Flutter/pull/990))
```
Comment (mtrezza) [inline packages/dart/CHANGELOG.md:5]:
```suggestion
* Add `ParseXFile` for cross-platform `XFile` support ([#990](https://github.com/parse-community/Parse-SDK-Flutter/pull/990))
```

**Review by mtrezza — APPROVED** [review]

**Comment by mtrezza** [comment]
I just noticed that the docs workflow was skipped; shouldn't docs be published?

https://github.com/parse-community/Parse-SDK-Flutter/actions/runs/8493279325/job/23267194136

**Comment by mbfakourii** [comment]
> I just noticed that the docs workflow was skipped; shouldn't docs be published?
> 
> https://github.com/parse-community/Parse-SDK-Flutter/actions/runs/8493279325/job/23267194136

I think this section had a problem in ci.

**Comment by mtrezza** [comment]
I tried to run it manually but it also wouldn't run. I'll look into it.

**Pull request merged**